pandas/_libs/lib.pyx

```diff
@@ -1511,6 +1517,21 @@ cpdef bint is_integer_array(ndarray values):
     return validator.validate(values)
 
 
+cdef class IntegerNaValidator(Validator):
+    cdef inline bint is_value_typed(self, object value) except -1:
+        return util.is_integer_object(value) or util.is_nan(value)
```

Comment:
I think you need a is_valid_null as well (e.g. see how PeriodValidator is done)

Reply:
I changed to `util.is_nan(value) and util.is_float_object(value)` instead